src/resolvers/project.ts

```diff
@@ -151,6 +268,58 @@ export const resolvers: Resolvers = {
         throw InternalServerError();
       }
     },
+
+    // Import project from an external data source
+    projectImport: async (_, {input}, context) => {
+      const reference = 'updateProject resolver';
+      try {
+        if (isAuthorized(context.token)) {
+          try {
+            const projectId = input.project.id;
+            const project = await Project.findById(reference, context, projectId);
+            if (!project) {
+              throw NotFoundError();
+            }
+
+            if (!hasPermissionOnProject(context, project)) {
+              throw ForbiddenError();
+            }
+
+            // Update project
+            const toUpdate = new Project(input.project);
+            await toUpdate.update(context);
+
+            // Update project funding
+            for (const fund of input.funding) {
+              const toCreate = new ProjectFunder(fund);
+              await toCreate.create(context, projectId);
+            }
+
+            // Update project contributors
+            for (const contrib of input.contributors) {
+              // Add project contributor
+              const toCreate = new ProjectContributor(contrib);
+              await toCreate.create(context, projectId);
+
+              // TODO: raise errors
```

Comment:
Adding project funders and contributors can fail, so do I need to combine all of the errors and return them as a part of the response? At the moment I'm returning the project object, which will only have its own errors.

Reply:
Yes, we will want to capture and report those errors as well. I've done something similar here: https://github.com/CDLUC3/dmsp_backend_prototype/blob/864312ac43052685baaf81c767133d323708e9a0/src/resolvers/contributor.ts#L103
You can combine all of the errors and add them to `project.addError('general', 'the errors messages')`

Reply:
Thanks @briri, I'll do that.